zebra-chain/src/block.rs

```diff
@@ -36,6 +38,28 @@ pub struct Block {
     pub transactions: Vec<std::sync::Arc<Transaction>>,
 }
 
+impl fmt::Display for Block {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut fmter = f.debug_struct("Block");
+        if let Some(height) = self.coinbase_height() {
+            fmter.field("height", &height);
+        }
+
+        fmter.field("hash", &DisplayToDebug(self.hash())).finish()
+    }
+}
+
+struct DisplayToDebug<T>(T);
+
+impl<T> fmt::Debug for DisplayToDebug<T>
+where
+    T: fmt::Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
```

Comment:
Nice!

I'm sure I did something similar somewhere else... but I can't seem to find it right now. Maybe I didn't end up needing it, so I deleted it.

Reply:
Ha, I had the exact same reaction when writing this. We probably should make a place to store little utils like this.

Reply:
We don't have a dedicated crate for them, but maybe `zebra-chain/fmt.rs` would be a good place for `DisplayToDebug` and `NoDebug`.

Reply:
I think this is maybe a question best answered by @hdevalence because I know he's got a firm idea of how code should be organized that I generally agree with but fail to predict.

Reply:
It would be nice to write that down somewhere - it could save us a bit of refactoring during reviews.

Reply:
o yes, that would be quite nice. Henry guide on code organization

Reply:
Maybe we should open another ticket for this kind of code movement?
It doesn't need to be part of this PR.

Reply:
I just talked to henry about it and only unresolved this so he could more easily find the comment, I'll leave it up to him if he wants to make an issue to track it.